### PR TITLE
Parsec: Don't require all items in list to be quoted if one must

### LINKED ISSUE #1520: #1475 user auth (step 2 extended) - user privilege groups
#1480 distinguishes between two user types: "authenticated" (with passphrase) and "public" (no passphrase).  Global/suite config can set the privilege level for public users, and authenticated users get full control privileges.  Public read-only is now an option, but is probably not advisable at a large site because of the potential for hammering suite daemons with GUI info requests.

The new connection validator receives the client user and host names (and UUID) as well as passphrase, so we easily could split both public and authenticated users into multiple groups according to users and/or hosts specified under the new config [authentication] sections.  So for instance, you could allow a specified subset of all users, or of users on a particular host, to have read-only access to a suite, and allow only "identity" information to others.  Or you could give a subset of authenticated (with passphrase) users read-only access.

This would be almost as good as the proposed final "user account" solution (step 3 of #1475) but it could be implemented very quickly (est. 1 hour after #1480) and would give us some freedom to think through the final solution carefully.

### LINKED ISSUE #5120: parsec: comma separated list items containing commas
Spotted in relation to #5118, the "supported" syntax for multiple parameters doesn't play nicely with the inherit configuration either.

For example this workflow:

```ini
[task parameters]                                                                    
    x = 1..2           
    y = 3..4           
                       
[scheduling]           
    [[dependencies]]   
        [[[R1]]]       
            graph = foo
                       
[runtime]              
    [[I, J]]           
    [[A<x,y>]]         
    [[a1<x,y>, a2<x,y>, a3<x,y>]]
        inherit = I, "A<x,y>", J 
```

Looks fine, but doesn't validate:

```
ListValueError: (type=list) [runtime][a1<x,y>, a2<x,y>, a3<x,y>]inherit = I, "A<x,y>", J - (names containing commas must be quoted(e.g. 'foo<m,n>'))
```

The parametrised item *has* been quoted as suggested by the error message, but it still fails validation.

I found a workaround, for some reason if one of the items in the list is quoted, all of the items must be, whether they need to be or not:

```diff
     [[a1<x,y>, a2<x,y>, a3<x,y>]]
-        inherit = I, "A<x,y>", J 
+        inherit = "I", "A<x,y>", "J" 
```

Should be a simple fix, probably in the "coerser".

### PR DESCRIPTION
**Pull request by wxtim**
Closes #5120

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [ ] Tests are included (or explain why tests are not needed).
  - [x] Unit tests added and refined.
  - [ ] Run `cylc config` against lots of workflows locally
- [ ] `CHANGES.md` entry included if this is a change that can affect users
- [ ] If this is a bug fix, PRs raised to both master and the relevant maintenance branch.

### CODE REVIEW

File: cylc/flow/parsec/validate.py [hunk 1]
```diff
@@ -582,27 +585,43 @@ def _unquoted_list_parse(cls, keys, value):
...
+            >>> list(ParsecValidator._unquoted_list_parse(
+            ...     None, 'I, "A<x,y>", J, "<a, b>B"'))
+            ['I', 'A<x,y>', 'J', '<a, b>B']
         """
-        # http://stackoverflow.com/questions/4982531/
-        # how-do-i-split-a-comma-delimited-string-in-python-except-
-        # for-the-commas-that-are
 
-        # First detect multi-parameter lists like <m,n>.
-        if cls._REC_MULTI_PARAM.search(value):
```
Comment (wxtim) [inline cylc/flow/parsec/validate.py:592]: Running this regex before attempting to parse the list will mean we either reject or pass the entire list.

File: cylc/flow/parsec/validate.py [hunk 2]
```diff
@@ -582,27 +584,42 @@ def _unquoted_list_parse(cls, keys, value):
         Examples:
             >>> list(ParsecValidator._unquoted_list_parse(None, '"1", 2'))
             ['1', '2']
```
Comment (MetRonnie) [inline cylc/flow/parsec/validate.py:586]: 😤
```
>>> list(ParsecValidator._unquoted_list_parse(None, '1,"2",3'))
['1', '2', '3']

>>> list(ParsecValidator._unquoted_list_parse(None, '1, "2", 3'))
['1', '"2"', '3']
```

**Review by MetRonnie — COMMENTED** [review]
Sorry, these 2 comments are nothing to do with your changes but I have been baffled by these

File: cylc/flow/parsec/validate.py (hunk 2 above)
Comment (hjoliver) [reply cylc/flow/parsec/validate.py:586]: We've discussed this somewhere recently, just trying to find evidence...

Comment (hjoliver) [reply cylc/flow/parsec/validate.py:586]: Probably in relation to #5118

**Pull request closed without merging**